pkg/server/home_workspaces.go

```diff
@@ -0,0 +1,656 @@
+	createClusterRole        func(ctx context.Context, lcluster logicalcluster.Name, cr *rbacv1.ClusterRole) error
+	createClusterRoleBinding func(ctx context.Context, lcluster logicalcluster.Name, crb *rbacv1.ClusterRoleBinding) error
+}
+
+func buildExternalClientsAccess(kubeClusterClient kubernetes.ClusterInterface, kcpClusterClient kcpclient.ClusterInterface) externalKubeClientsAccess {
+	return externalKubeClientsAccess{
+		createClusterRole: func(ctx context.Context, workspace logicalcluster.Name, cr *rbacv1.ClusterRole) error {
+			_, err := kubeClusterClient.Cluster(workspace).RbacV1().ClusterRoles().Create(ctx, cr, metav1.CreateOptions{})
+			return err
+		},
+		createClusterRoleBinding: func(ctx context.Context, workspace logicalcluster.Name, crb *rbacv1.ClusterRoleBinding) error {
+			_, err := kubeClusterClient.Cluster(workspace).RbacV1().ClusterRoleBindings().Create(ctx, crb, metav1.CreateOptions{})
+			return err
+		},
+		createClusterWorkspace: func(ctx context.Context, workspace logicalcluster.Name, cw *tenancyv1alpha1.ClusterWorkspace) error {
+			_, err := kcpClusterClient.Cluster(workspace).TenancyV1alpha1().ClusterWorkspaces().Create(ctx, cw, metav1.CreateOptions{})
+			return err
+		},
+		getClusterWorkspace: func(ctx context.Context, workspace logicalcluster.Name, name string) (*tenancyv1alpha1.ClusterWorkspace, error) {
+			return kcpClusterClient.Cluster(workspace).TenancyV1alpha1().ClusterWorkspaces().Get(ctx, name, metav1.GetOptions{})
+		},
+	}
+}
+
+type localInformersAccess struct {
+	getClusterWorkspace      func(logicalcluster.Name) (*tenancyv1alpha1.ClusterWorkspace, error)
+	searchClusterRole        func(lcluster logicalcluster.Name, name string) (bool, error)
+	searchClusterRoleBinding func(lcluster logicalcluster.Name, name string) (bool, error)
+	synced                   func() bool
+}
+
+func buildLocalInformersAccess(kubeSharedInformerFactory coreexternalversions.SharedInformerFactory, kcpSharedInformerFactory kcpexternalversions.SharedInformerFactory) localInformersAccess {
+	clusterWorkspaceInformer := kcpSharedInformerFactory.Tenancy().V1alpha1().ClusterWorkspaces().Informer()
+	crInformer := kubeSharedInformerFactory.Rbac().V1().ClusterRoles().Informer()
+	crbInformer := kubeSharedInformerFactory.Rbac().V1().ClusterRoleBindings().Informer()
+	clusterWorkspaceLister := kcpSharedInformerFactory.Tenancy().V1alpha1().ClusterWorkspaces().Lister()
+	crLister := kubeSharedInformerFactory.Rbac().V1().ClusterRoles().Lister()
+	crbLister := kubeSharedInformerFactory.Rbac().V1().ClusterRoleBindings().Lister()
+
+	return localInformersAccess{
+		getClusterWorkspace: func(logicalCluster logicalcluster.Name) (*tenancyv1alpha1.ClusterWorkspace, error) {
+			parentLogicalCluster, workspaceName := logicalCluster.Split()
+			return clusterWorkspaceLister.Get(clusters.ToClusterAwareKey(parentLogicalCluster, workspaceName))
+		},
+		searchClusterRole: func(workspace logicalcluster.Name, name string) (bool, error) {
```

Comment:
am still confused about these function name. Make them normal informer getters. Would be one (two) surprise(s) less.

Reply:
done